src/pickup.cpp

```diff
@@ -169,6 +179,34 @@ interact_results interact_with_vehicle( vehicle *veh, const tripoint &pos,
             veh_tool( "hotplate" );
             return DONE;
 
+        case USE_WASHMACHINE:
+            if( g->u.crafting_inventory().has_charges( "detergent", 5 ) ) {
+                detergent_is_enough = true;
+            }
+
+            for( auto e : veh->get_parts( "WASHING_MACHINE" ) ) {
+                if( e->enabled ) {
+                    e->enabled = false;
+                    veh->tags.insert( "manual_stop" );
+                    add_msg( m_neutral, _( "You turn the washing machine off before it's finished the program, and open its lid." ) );
+                } else if( veh->fuel_left( "water" ) < 24 ) {
+                    add_msg( m_neutral, _( "You need 24 charges of water to fill the washing machine." ) );
+                } else if( !detergent_is_enough ) {
+                    add_msg( m_neutral, _( "You need 5 charges of detergent for the washing machine." ) );
+                } else {
+                    e->enabled = true;
+                    veh->drain( "water", 24 );
+
+                    std::vector<item_comp> detergent;
+                    detergent.push_back( item_comp( "detergent", 5 ) );
+                    g->u.consume_items( detergent );
+
+                    veh->tags.erase( "manual_stop" );
+                    add_msg( m_neutral, _( "You close the lid of the washing machine, and turn it on.  The washing machine is being filled with soapy water." ) );
+                }
+            }
```

Comment:
But it is already in `interact_with_vehicle` in `pickup.cpp`. And it indeed working by interacting with it directly, not through controls.